src/fmu/dataio/_objectdata_provider.py

```diff
@@ -200,10 +206,10 @@ def _validate_get_ext(fmt, subtype, validator):
 
         return validator.get(fmt, None)
 
-    def _derive_objectdata(self):
+    def _derive_objectdata(self) -> dict:
```

Comment:
```suggestion
    def _derive_objectdata(self) -> dict[str, Any]:
```
Still too permissive maybe, but less so